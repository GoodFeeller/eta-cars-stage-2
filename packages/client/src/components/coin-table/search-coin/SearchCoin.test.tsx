import { FunctionComponent, useState } from 'react'
import SearchCoin from './SearchCoin'
import { render, screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

const TestComponent: FunctionComponent = () => {
  const [search, setSearch] = useState<string>('')
  return (
    <div>
      <SearchCoin search={search} setSearch={setSearch} />
      <article>{search}</article>
    </div>
  )
}
describe('SearchCoin test', () => {
  test('Snapshot test', () => {
    render(<TestComponent />)
    const input = screen.queryByTestId('search')
    expect(input).toMatchSnapshot()
  })
  test('Snapshot `test` search test', async () => {
    await act(async () => render(<TestComponent />))
    const input = screen.queryByTestId('search') as HTMLInputElement
    userEvent.type(input, 'bitcoin')
    expect(input).toMatchSnapshot()
  })
  test('Input test', async () => {
    await act(async () => render(<TestComponent />))
    const input = screen.queryByTestId('search') as HTMLInputElement
    userEvent.type(input, 'bitcoin')
    const search = screen.queryByRole('article') as HTMLElement
    expect(input.value).toContain('bitcoin')
    expect(search.textContent).toContain('bitcoin')
  })
})
