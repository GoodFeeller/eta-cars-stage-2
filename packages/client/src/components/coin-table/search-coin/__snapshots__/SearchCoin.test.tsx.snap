// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCoin test Snapshot \`test\` search test 1`] = `
<input
  class="searchBody"
  data-testid="search"
  placeholder="Coin name..."
  type="text"
  value="bitcoin"
/>
`;

exports[`SearchCoin test Snapshot test 1`] = `
<input
  class="searchBody"
  data-testid="search"
  placeholder="Coin name..."
  type="text"
  value=""
/>
`;
